### PR TITLE
Report memory usage by memtable insert hints map.

### PR DESCRIPTION
**Pull request by yiwu-arbug**
Summary:
It is hard to measure acutal memory usage by std containers. Even
providing a custom allocator will miss count some of the usage. Here we
only do a wild guess on its memory usage.

Test Plan:
Have a piece of custom code to open a DB, insert multiple keys with hint
and verify the number is report through
DB::Properties::kCurSizeAllMemTables property.

### CODE REVIEW

**Comment by yiwu-arbug** [comment]
use `port::kMaxSizet` instead of `std::numeric_limits<size_t>::max()`.

**Comment by yiwu-arbug** [comment]
Fix miscount of bucket size.

File: util/memory_usage.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+template <class Key, class Value, class Hash>
+size_t ApproximateMemoryUsage(
+    const std::unordered_map<Key, Value, Hash>& umap) {
+  typedef std::unordered_map<Key, Value, Hash> Map;
+  return sizeof(umap) +
+      // Size of all items plus a next pointer for each item.
+      (sizeof(typename Map::value_type) + sizeof(void*)) * umap.size() +
+      // Size of hash buckets.
+      umap.bucket_count() * sizeof(void*);
+}
```
Comment (siying) [inline util/memory_usage.h]: I'm not sure how accurate it is. Just assume it is better than nothing.

Comment (yiwu-arbug) [inline util/memory_usage.h]: The estimation is from reading libstdc++'s comment:
https://github.com/gcc-mirror/gcc/blob/master/libstdc%2B%2B-v3/include/bits/hashtable.h#L102-L113
I guess different implementation would use a similar data structure.

Comment (yiwu-arbug) [inline util/memory_usage.h]: And yeah, it is only a better than nothing estimation.

**Comment by yiwu-arbug** [comment]
Will fix db_properties_test.

**Comment by yiwu-arbug** [comment]
Fix db_properties_test.

**Comment by yiwu-arbug** [comment]
Fix lite build.

**Pull request closed without merging**